docs/architecture/component_templates.md

```diff
@@ -0,0 +1,134 @@
+# Account component templates
+
+An account component template provides a general description of an account component. It encapsulates all the information needed to initialize and manage the component.
+
+Specifically, a template specifies a component's **metadata** and its **code**.
+
+Once defined, a component template can be instantiated as account components, which can then be grouped to form the account's `Code`.
+
+## Component code
+
+The component template’s code defines a library of functions that operate on the specified storage layout.
+
+## Component metadata
+
+The component metadata describes the account component entirely: its name, description, version, and storage layout.
+
+The storage layout can optionally define specific initial values for each of the slots. Alternatively, placeholders can be utilized to identify values that should be provided at the moment of instantiation.
+
+### TOML specification
+
+The component metadata can be defined using TOML. Below is an example specification:
+
+```toml
+name = "Fungible Faucet"
+description = "This component displays valid values for component templates"
+version = "1.0.0"
+supported-types = ["FungibleFaucet"]
+
+[[storage]]
+name = "token_metadata"
+description = "Contains metadata about the token associated to the faucet account. The metadata is formed by three fields: max supply, the token symbol and the asset's decimals"
+slot = 0
+value = [
+    { type = "felt", name = "max_supply", description = "Maximum supply of the token in base units" },
+    { type = "tokensymbol", value = "TST" },
+    { type = "u8", name = "decimals", description = "Number of decimal places for converting to absolute units", value = "10" },
+    { value = "0x0" }
+]
+
+[[storage]]
+name = "owner_public_key"
+description = "This is a value placeholder"
+slot = 1
+type = "auth::rpo_falcon512::pub_key"
+
+[[storage]]
+name = "map_storage_entry"
+slot = 2
+values = [
+    { key = "0x1", value = ["0x0", "249381274", "998123581", "124991023478"] },
+    { key = "0xDE0B1140012A9FD912F18AD9EC85E40F4CB697AE", value = { name = "value_placeholder", description = "This value will be defined at the moment of instantiation", type = "word" } }
+]
+```
```

Comment:
Is this intentionally missing a multi-slot example? I think for the docs it would be useful to have an "exhaustive example", i.e. an example that shows the entire range of possibilities. That way we can refer to each variant in the specification below.

Reply:
Yeah, I intentionally left it out, though I debated adding one. I was wondering if it would not add more noise than anything else, since there currently is no typing for multi-slot values which means that essentially you are just defining a list of single words. In any case, for completeness, I will add one.